docs/howto/migrate/migration-guide.md

```diff
@@ -0,0 +1,1163 @@
+
+![ZIO Streams 2.x](assets/zio-streams-2.x.svg)
+
+Pipelines are basically an abstraction for composing a bunch of operations together that can be later applied to a stream. For example, we can create a pipeline that reads bytes, decodes them to the UTF-8 and splits the lines, and then splits on commas. So this is a very simple CSV parsing pipeline which we can later use with another stream to pipe into. 
+
+| ZIO Streams 1.x | ZIO Streams 2.x                  |
+|-----------------|----------------------------------|
+|                 | `ZChannel`                       |
+| `ZStream`       | `ZStream` (backed by `ZChannel`) |
+| `ZSink`         | `ZSink` (backed by `ZChannel`)   |
+| `ZTransducer`   | `ZPipeline`                      |
+
+## ZIO Services
+
+There are two significant changes in ZIO Services:
+
+1. All ZIO services moved to the `zio` package:
+
+    | ZIO 1.x                 | ZIO 2.x                      |
+    |-------------------------|------------------------------|
+    | `zio.blocking.Blocking` | [Removed](#blocking-service) |
+    | `zio.clock.Clock`       | `zio.Clock`                  |
+    | `zio.console.Console`   | `zio.Console`                |
+    | `zio.random.Random`     | `zio.Random`                 |
+    | `zio.system.System`     | `zio.System`                 |
+
+    And their live implementations renamed and moved to a new path:
+
+    | ZIO 1.x                    | ZIO 2.x                   |
+    |----------------------------|---------------------------|
+    | `zio.Clock.Service.live`   | `zio.Clock.ClockLive`     |
+    | `zio.Console.Service.live` | `zio.Console.ConsoleLive` |
+    | `zio.System.Service.live`  | `zio.System.SystemLive`   |
+    | `zio.Random.Service.live`  | `zio.Random.RandomLive`   |
+
+
+2. In ZIO 2.0 all type aliases like `type Logging = Has[Logging.Service]` removed. So we should explicitly use `Has` wrappers when we want to specify dependencies on ZIO services.
+
+So instead of writing `ZLayer[Console with Clock, Nothing, ConsoleLogger]`, we should write `ZLayer[Has[Console] with Has[Clock], Nothing, Has[ConsoleLogger]]`, or when accessing services instead of `ZIO.service[Console.Service]` we also now do `ZIO.service[Console]`.
+
+### Blocking Service
+
+Since there is rarely a need to use a separate blocking thread pool, ZIO 2.0 created _one global blocking pool_ removed the Blocking service from `ZEnv` and the built-in services.
+
+All blocking operations were moved to the `ZIO` data type:
+
+| ZIO 1.x                   | ZIO 2.x |
+|---------------------------|---------|
+| `zio.blocking.Blocking.*` | `ZIO.*` |
+
+We can also provide a user-defined blocking executor in ZIO 2.x with the `Runtime#withBlockingExecutor` operator that constructs a new `Runtime` with the specified blocking executor.
```

Comment:
we should mention that we now have `ZIO.attemptBlocking`, `ZIO.attemptBlockingIO` etc. instead

Reply:
Thank you! Fixed.